In [1]:
import sqlite3
import csv
from pprint import pprint

In [2]:
sqlite_file = "C:\Users\James2SxyBoogaloo\Google Drive\Data\P3 -  Data Wrangling\database.db"
#connect to database
conn =sqlite3.connect(sqlite_file)

In [5]:
c = conn.cursor()

## Top ten cuisines

In [6]:
query = '''
SELECT value, count(*) as num \
FROM nodes_tags \
WHERE key ='cuisine' \
GROUP BY value \
ORDER BY num desc \
LIMIT 10;
'''

def cuisine():
    c.execute(query)
    results = c.fetchall()
    print results

cuisine()

[(u'chinese', 70), (u'indian', 66), (u'italian', 49), (u'sandwich', 40), (u'fish_and_chips', 31), (u'pizza', 21), (u'burger', 18), (u'coffee_shop', 18), (u'regional', 16), (u'asian', 10)]


Chinese narrowly edges out Indian as glasgow's preferred cuisine.

## Top ten amenities

In [7]:
query2 = '''
SELECT value, count(*) as num \
FROM nodes_tags \
WHERE key ='amenity' \
GROUP BY value \
ORDER BY num desc \
LIMIT 10;
'''

def amenity():
    c.execute(query2)
    results = c.fetchall()
    print results

amenity()

[(u'bicycle_parking', 556), (u'post_box', 458), (u'fast_food', 307), (u'restaurant', 268), (u'pub', 255), (u'cafe', 229), (u'telephone', 166), (u'atm', 120), (u'bench', 115), (u'recycling', 98)]


Glasgow benefits from a surfeit of bicycle parking.

## Top ten shops

In [26]:
query3 = '''
SELECT value, count(*) as num \
FROM nodes_tags \
WHERE key ='shop' \
GROUP BY value \
ORDER BY num desc \
LIMIT 10;
'''

def shop():
    c.execute(query3)
    results = c.fetchall()
    print results

shop()

[(u'convenience', 231), (u'hairdresser', 119), (u'clothes', 65), (u'supermarket', 65), (u'bookmaker', 54), (u'newsagent', 50), (u'beauty', 40), (u'bakery', 32), (u'yes', 27), (u'butcher', 24)]


As can be seen, the overwhelming majority of shops in glasgow are convenience stores. There is a surprisingly low number of bookmakers - as this is a proxy for deprivation it would be interesting to explore this further, perhaps by areacode. Outside my flat there are 3 bookkeepers within a 60 second walk.

Note that 'Yes' is getting returned as a type of shop - given that the existence of the shop key implies that a shop exists, this is a potential further opportunity for data cleaning.

## Areacodes with the most Bookmakers

In [27]:
query4 = '''
SELECT nodes_tags.value, COUNT(*) as num
FROM nodes_tags
    JOIN (SELECT DISTINCT(id) FROM nodes_tags WHERE value= 'bookmaker') i 
    ON nodes_tags.id=i.id
WHERE nodes_tags.key = 'Areacode'
GROUP BY nodes_tags.value
ORDER BY num DESC
LIMIT 10;
'''

def ACB():
    c.execute(query4)
    results = c.fetchall()
    print results

ACB()

[(u'G1', 3), (u'G2', 1), (u'G4', 1)]


This is less than ideal - our query returns only 5 results.

An examination of the CSV indicates that the vast majority of the bookmakers listed in the data do not have an associated postcode: as a result neither do they have an Areacode. - This isn't a cleaning issue however.

In the interest of getting a meaningful output lets investigate restaurants - perhaps these will have fully completed postcodes.

## Areacodes with the most restaurants


In [28]:
query5 = '''
SELECT nodes_tags.value, COUNT(*) as num
FROM nodes_tags
    JOIN (SELECT DISTINCT(id) FROM nodes_tags WHERE value= 'restaurant') i 
    ON nodes_tags.id=i.id
WHERE nodes_tags.key = 'Areacode'
GROUP BY nodes_tags.value
ORDER BY num DESC
LIMIT 10;
'''

def AMR():
    c.execute(query5)
    results = c.fetchall()
    print results

AMR()

[(u'G1', 13), (u'G2', 7), (u'G3', 6), (u'G12', 3), (u'G31', 3), (u'G4', 2), (u'G41', 2), (u'G53', 2), (u'G20', 1), (u'G33', 1)]


Once again we are challenged by the incompleteness of the data. Despite the previous count showing a total of no less than 268 restaurants, our top ten Areacodes only account for a small proportion of this total. However assuming that the likelihood of the restuarant not having an associated postcode (and thus Areacode) is unaffected by location, we can still conclude that G1 has a significant proportion of the total.
